### DISCUSSION TITLE
I made an open-source Android transcription keyboard using Whisper AI. You can dictate with auto punctuation and translation to many languages. :)

### DISCUSSION THREAD

**Discussion by DevEmperor** [openai/whisper#2227] [Show and tell]
![dictate_ad](https://github.com/openai/whisper/assets/56255079/312a5b59-39b0-4405-a640-a034410b1b60)

Dictate is an easy-to-use keyboard for transcribing and dictating. The app uses OpenAI Whisper in the background, which supports extremely accurate results for many different languages with punctuation and auto translation using GPT-4 Omni.

You can download the app from Google Play Store:
https://play.google.com/store/apps/details?id=net.devemperor.dictate

Here you can see it in action:
https://www.youtube.com/watch?v=PSvLRnHYleg

And this is the repository with the source code:
https://github.com/DevEmperor/Dictate

**Comment by ipstone** [comment]
nice interface! But my question is why sending the audio to openAI for the transcribing? why not just do on device transcription, such as this one:
  https://play.google.com/store/apps/details?id=net.devemperor.dictate&pli=1
  or this one:
  https://play.google.com/store/apps/details?id=com.signalmaster.asr&hl=en_US

For the second one, the code is open at :
https://github.com/vilassn/whisper_android

If your keyboard is using these local transcription engine without sending audio out, it will be a really nice/popular app!

**Reply by KanyantaM** [reply]
Perhaps the accuracy of on device transcription isn't that good. It really depends on the quality of your device

**Comment by DevEmperor** [comment]
Hey, thanks for this great tip. However, I want to design the app for everyone and most smartphones are not that powerful to run a local Whisper model. Also, I want to achieve quick results and the API may cost some money and require an internet connection, but it is fast and reliable.
I will keep this feature in mind, but currently, I sadly don't have time to implement it.
Thanks anyway for your idea. :)

**Reply by fire17** [reply]
Hopefully both options will be available.. run a small bench to determine which will be default (if sample takes less than x time to transcribe use local by default, else use api) 
Got here cause I'm looking for an open source keyboard with ai abilities.... Not just transcribing but also text completion (copilot style) and other features

**Reply by Benko111** [reply]
> Hey, thanks for this great tip. However, I want to design the app for everyone and most smartphones are not that powerful to run a local Whisper model. Also, I want to achieve quick results and the API may cost some money and require an internet connection, but it is fast and reliable. I will keep this feature in mind, but currently [...]

Well, by now the HF-Community has a pretty broad selection of optimitzed (quantisized and distilled) on-device Whisper models, some of them weighing in at under 100MB iirc. So, if device specs are actually your concern that can be helped. I've read some of your users comments and indeed they praise stability and speed. Offering an aditional privacy-adhering offline option will certainly be a bonus, or an additional offline-version, etc. whatever... I'm sure this will make some additional conversions. Good library with some models: 

https://github.com/Vaibhavs10/insanely-fast-whisper

Or finetune your own Whisper models on consumer hardware, resulting in very small but just as capable models:

https://github.com/Vaibhavs10/fast-whisper-finetuning
